docs/dev_tools/build_profiles.md

```diff
@@ -0,0 +1,21 @@
+# Build Profiles
+mbed OS 5 supports three primary build profiles: *default*, *debug* and *small*. The Online 
+Compiler uses the *default* profile. When building from mbed CLI,
+you can select a profile by adding the ```--profile <profile>```
+flag. You can specify custom user defined profiles by giving the path
+to the profile.
+
+## Default profile
+* Small and fast code.
+* Full error information - e.g. asserts have file name and line number.
+* Hard to follow code flow when using a debugger.
+
+## Debug profile
+* Easy to step through code with a debugger.
```

Comment:
This comes first here and last in the other two; as a reader I'd like a consistent order for topics